src/Client/ShopClient.php

```diff
@@ -58,7 +58,7 @@ private function createToken(): Credentials
 
         $authRequest = new Request(
             'POST',
-            new Uri(self::AUTHENTICATION_ROUTE),
+            $this->shop->getUrl() . self::AUTHENTICATION_ROUTE,
```

Comment:
The problem here is that we don't know how the `base_uri` looks like. The docs (https://docs.guzzlephp.org/en/stable/quickstart.html#creating-a-client) describe a lot of possible cases when and how the base_uri is used. So there is a difference if the shop URL ends with a slash. Then the concatenation will not work.

Reply:
We should try the following:
1. ensure that the base_uri ends with a slash when passed to the guzzle client
2. Use relative URLs without preceding slash (e.g. api/oauth/token)